### PR TITLE
add leaderboard caching to gulag

### PR DESCRIPTION
**Pull request by cmyui**
yet another attempt to carve out an implementation idea for leaderboard caching.

not sure whether or not this will be merged, but it's definitely providing more insight into how it'll work.

### CODE REVIEW

File: app/api/domains/map.py [hunk 1]
```diff
@@ -5,7 +5,7 @@
 from fastapi.responses import RedirectResponse
 
 
-router = APIRouter(prefix="/map", tags=["Beatmaps"])
+router = APIRouter(tags=["Beatmaps"])
```
Comment (cmyui) [inline app/api/domains/map.py:8]: unrelated bugfix i probably should have pushed to master

fixes b.ppy.sh/*

**Comment by cmyui** [comment]
this really needs a rebase

**Comment by cmyui** [comment]
i have not yet given up. bancho.py will get a leaderboard cache eventually! lol

**Comment by tsunyoku** [comment]
i will write one, soon!

**Pull request closed without merging**